lldb/source/Plugins/OperatingSystem/SwiftTasks/OperatingSystemSwiftTasks.h

```diff
@@ -49,10 +50,18 @@ class OperatingSystemSwiftTasks : public OperatingSystem {
   /// If a thread for task_id had been created in the last stop, return it.
   /// Otherwise, create a new MemoryThread for it.
   lldb::ThreadSP FindOrCreateSwiftThread(ThreadList &old_thread_list,
-                                         uint64_t task_id);
+                                         uint64_t task_id,
+                                         std::optional<std::string> task_name);
 
-  /// Find the Task ID of the task being executed by `thread`, if any.
-  std::optional<uint64_t> FindTaskId(Thread &thread);
+  /// Find the address of the Task being executed by `thread`, if any.
+  std::optional<lldb::addr_t> FindTaskAddress(Thread &thread);
+
+  /// Find the ID of the Task at the given address, if any.
+  std::optional<uint64_t> FindTaskId(lldb::addr_t task_addr);
+
+  /// Find the (optional) name of the Task at the given address, if any.
+  std::optional<std::string> FindTaskName(lldb::addr_t task_addr,
+                                          Process *process);
```

Comment:
This method doesn't need to me a member, it can be declared and defined inside the cpp file